src/server/pfs/drive/driver.go

```diff
@@ -18,6 +19,17 @@ import (
 	"google.golang.org/grpc"
 )
 
+func newPermissionError(repo string, commitID string) error {
+	return fmt.Errorf("commit %s/%s has already been finished", repo, commitID)
+}
+
+func IsPermissionError(err error) bool {
+	if readOnly := strings.Contains(err.Error(), "has already been finished"); readOnly {
+		return true
+	}
+	return false
+}
+
```

Comment:
Ah, yea.

The match sting 'has already been finished' is not the most descriptive thing, so I was hoping to make the code around the matching more descriptive ... but I can see how that's not really the case right now either.

I'll inline as you suggested.